Hive.NET.Core/Components/Bee.cs

```diff
@@ -11,16 +11,25 @@ public Bee()
         IsWorking = false;
     }
 
-    public async Task DoWork(Task unitOfWork, BeeFinishedWorkCallback beeCallback)
+    public async Task DoWork(BeeWorkItem unitOfWork, 
+        BeeFinishedWorkCallback beeCallback)
     {
-        IsWorking = true;
-        
-        unitOfWork.Start();
-        await unitOfWork;
-        
-        IsWorking = false;
+        try
+        {
+            IsWorking = true;
+
+            unitOfWork.task.Start();
+            await unitOfWork.task;
+
+            IsWorking = false;
 
-        beeCallback(this);
+            beeCallback(this);
```

Comment:
Maybe add some log to the default callback so it would notify that it has indeed executed